src/sig_stfl/lms/sig_stfl_lms_functions.c

```diff
@@ -264,3 +350,139 @@ int oqs_sig_stfl_lms_verify(const uint8_t *m, size_t mlen,
 	}
 }
 
+void oqs_secret_lms_key_free(OQS_SIG_STFL_SECRET_KEY *sk) {
+	if (sk == NULL) {
+		return;
+	}
+
+	//TODO: cleanup lock_key
+
+	if (sk->sig) {
+		OQS_MEM_insecure_free(sk->sig);
+		sk->sig = NULL;
+	}
+
+	if (sk->secret_key_data) {
+		oqs_lms_key_data *key_data = (oqs_lms_key_data *)sk->secret_key_data;
+		if (key_data) {
+			OQS_MEM_secure_free(key_data->sec_key, key_data->len_sec_key);
+			key_data->sec_key = NULL;
+
+			OQS_MEM_secure_free(key_data->aux_data, key_data->len_aux_data);
+		}
+
+		OQS_MEM_insecure_free(key_data->data);
+		OQS_MEM_insecure_free(key_data);
+		sk->secret_key_data = NULL;
+	}
+}
+
+/*
+ * Convert LMS secret key object to byte string
+ * Writes secret key + aux data if present
+ */
+size_t oqs_serialize_lms_key(OQS_SIG_STFL_SECRET_KEY *sk,  uint8_t **sk_key) {
```

Comment:
```suggestion
size_t oqs_serialize_lms_key(const OQS_SIG_STFL_SECRET_KEY *sk,  uint8_t **sk_key) {
```

Seems like we can make this `const`.